### ISSUE TITLE
widget.Label.SetText() stoped working in 2.1.1

### ISSUE THREAD

**Issue by GGCristo** [fyne-io/fyne#2581]
### Describe the bug:
In fyne version 2.1.1 desktop.Hoverable stop working after a while. It works in 2.1.0

### Screenshots:
![showBug](https://user-images.githubusercontent.com/58046649/138557578-1ee6f773-fa4a-4916-8561-c858c5230ee8.gif)

As you can see it stop working after a while. Sometimes it takes longer to break.

### Device (please complete the following information):
 - **OS:** Linux, Ubuntu, X11
 - **Go version:** 1.17 
 - **Fyne version:** 2.1.1

**Comment by GGCristo** [comment]
Ok this is a minimal reproduction
```go
package main

import (
	"fmt"
	"strconv"

	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/driver/desktop"
	"fyne.io/fyne/v2/widget"
)
type MyWidget struct {
	widget.Label
  number int
}

func newMyWidget() *MyWidget {
  myWidget := &MyWidget{}
  myWidget.ExtendBaseWidget(myWidget)
  return myWidget
}
func (self *MyWidget) MouseIn(mouse *desktop.MouseEvent) {
  self.number++
  self.SetText(strconv.Itoa(self.number))
  fmt.Println("Mouse in")
}

// MouseMoved is a hook that is called if the mouse pointer moved over the element.
func (self *MyWidget) MouseMoved(mouse *desktop.MouseEvent) {
  self.number++
  self.SetText(strconv.Itoa(self.number))
  fmt.Println("Mouse moved")
}

// MouseOut is a hook that is called if the mouse pointer leaves the element.
func (ourimage *MyWidget) MouseOut() {
  ourimage.number = 0
  ourimage.SetText(strconv.Itoa(ourimage.number))
  fmt.Println("Mouse out")
}

func main() {
	a := app.New()
	w := a.NewWindow("vision-go")
  w.SetContent(newMyWidget())
  w.Resize(fyne.NewSize(500, 500))
  w.ShowAndRun()
}
```
This works in v2.1.0 but not in v2.1.1

**Comment by andydotxyz** [comment]
I tested on the `release/v2.1.x` branch and could not reproduce, so I think we have already fixed it ready for `v2.1.2`
